ec2/ec2_route.go

```diff
@@ -15,51 +15,80 @@
 package ec2
 
 import (
-	"context"
-	"database/sql"
-	"fmt"
-	"net/http"
-	"strings"
 	"time"
+	"errors"
+	"net/http"
+	"database/sql"
 
-	"github.com/trackit/jsonlog"
-	"gopkg.in/olivere/elastic.v5"
-
-	"github.com/trackit/trackit-server/aws/ec2"
 	"github.com/trackit/trackit-server/db"
-	"github.com/trackit/trackit-server/es"
-	"github.com/trackit/trackit-server/routes"
 	"github.com/trackit/trackit-server/users"
+	"github.com/trackit/trackit-server/routes"
 )
 
-// ec2QueryParams will store the parsed query params
-type ec2QueryParams struct {
-	accountList []string
-	indexList   []string
-}
+type (
+	// ec2QueryParams will store the parsed query params
+	ec2QueryParams struct {
+		accountList []string
+		indexList   []string
+	}
 
-// ec2HistoryQueryParams will store the parsed query params
-type ec2HistoryQueryParams struct {
-	accountList []string
-	indexList   []string
-	date        time.Time
-}
+	// ec2HistoryQueryParams will store the parsed query params
+	ec2HistoryQueryParams struct {
+		accountList []string
+		indexList   []string
+		date        time.Time
+	}
 
-// ec2QueryArgs allows to get required queryArgs params
-var ec2QueryArgs = []routes.QueryArg{
-	routes.AwsAccountsOptionalQueryArg,
-}
+	// ec2UnusedQueryParams will store the parsed query params
+	ec2UnusedQueryParams struct {
+		accountList []string
+		indexList   []string
+		date        time.Time
+		count       int
+		by          string
+	}
+)
 
-// ec2QueryArgs allows to get required queryArgs params
-var ec2HistoryQueryArgs = []routes.QueryArg{
-	routes.AwsAccountsOptionalQueryArg,
-	routes.QueryArg{
-		Name:        "date",
-		Type:        routes.QueryArgDate{},
-		Description: "Date with year and month. Format is ISO8601",
-		Optional:    false,
-	},
-}
+var (
+	// ec2QueryArgs allows to get required queryArgs params
+	ec2QueryArgs = []routes.QueryArg{
+		routes.AwsAccountsOptionalQueryArg,
+	}
+
+	// ec2HistoryQueryArgs allows to get required queryArgs params
+	ec2HistoryQueryArgs = []routes.QueryArg{
+		routes.AwsAccountsOptionalQueryArg,
+		routes.QueryArg{
```

Comment:
You already used a date param in your EC2 cost PR, please create a common query arg and use it everywhere